types/react-autocomplete/index.d.ts

```diff
@@ -143,8 +143,27 @@ declare namespace Autocomplete {
         open?: boolean;
         debug?: boolean;
     }
+
+    interface State {
```

Comment:
Ideally, state is indeed internal to the component. However, there are escape hatches for when you need to access the state of a child component using [refs](https://reactjs.org/docs/refs-and-the-dom.html). When using a third library, this can be a necessity to work around certain limitations. Here's a simplified portion of my code as an example:

```tsx
class SearchForm extends Component<Props, State> {
  autoCompleteRef = React.createRef<AutoComplete>();

  handleOnKeyDown: KeyboardEventHandler<HTMLFormElement> = (event) => {
    const autoCompleteState: AutoComplete.State = this.autoCompleteRef.current!.state;
    const { highlightedIndex } = autoCompleteState;
    const isFirstSuggestionSelected = highlightedIndex === 0;

    if (event.key === 'ArrowUp' && isFirstSuggestionSelected) {
      // do something to hide suggestions menu
    }
  };

  render() {
    return (
      <form onKeyDown={this.handleOnKeyDown}>
        <AutoComplete ref={this.autoCompleteRef} {...otherProps} />
      </form>
    );
  }
}
```

We are not exposing anything that isn't already exposed. React already allows us to read the state of the child component, all we're doing is annotating it with the correct type. If we don't add the `State` type, then the `autoCompleteState` object will have type `any`.

Reply:
Ok, fair enough... just wanted to make sure we weren't unnecessarily exposing state.